modules/contact/src/constraints/ComputeFrictionalForceLMMechanicalContact.C

```diff
@@ -0,0 +1,215 @@
+
+ComputeFrictionalForceLMMechanicalContact::ComputeFrictionalForceLMMechanicalContact(
+    const InputParameters & parameters)
+  : ComputeWeightedGapLMMechanicalContact(parameters),
+    _c_t(getParam<Real>("c_t")),
+    _friction_var(getVar("friction_lm", 0)),
+    _secondary_x_dot(_secondary_var.adUDot()),
+    _primary_x_dot(_primary_var.adUDotNeighbor()),
+    _secondary_y_dot(adCoupledDot("disp_y")),
+    _primary_y_dot(adCoupledNeighborValueDot("disp_y")),
+    _epsilon(1.0e-7),
+    _mu(getParam<Real>("mu"))
+{
+#ifndef MOOSE_GLOBAL_AD_INDEXING
+  mooseError(
+      "ComputeFrictionalForceLMMechanicalContact relies on use of the global indexing container "
+      "in order to make its implementation feasible");
+#endif
+
+  if (!getParam<bool>("use_displaced_mesh"))
+    paramError("use_displaced_mesh",
+               "'use_displaced_mesh' must be true for the "
+               "ComputeFrictionalForceLMMechanicalContact object");
+}
+
+void
+ComputeFrictionalForceLMMechanicalContact::computeQpProperties()
+{
+  // Get the value of _qp_gap
+  ComputeWeightedGapLMMechanicalContact::computeQpProperties();
+
+  // Compute the value of weighted tangential velocity
+  // Build the velocity vector
+
+  if (_has_primary)
+  {
+    ADRealVectorValue relative_velocity(_secondary_x_dot[_qp] - _primary_x_dot[_qp],
+                                        _secondary_y_dot[_qp] - _primary_y_dot[_qp],
+                                        0);
+    // Add derivative information
+    relative_velocity(0).derivatives() =
+        _secondary_x_dot[_qp].derivatives() - _primary_x_dot[_qp].derivatives();
+    relative_velocity(1).derivatives() =
+        _secondary_y_dot[_qp].derivatives() - _primary_y_dot[_qp].derivatives();
+
+    // Get the component in the tangential direction
+    _qp_tangential_velocity = relative_velocity * _tangents[_qp][0];
+  }
+  else
+    _qp_tangential_velocity = std::numeric_limits<Real>::quiet_NaN();
+}
+
+void
+ComputeFrictionalForceLMMechanicalContact::computeQpIProperties()
+{
+  // Get the _node_to_weighted_gap map
+  ComputeWeightedGapLMMechanicalContact::computeQpIProperties();
+
+  const auto * const node = _lower_secondary_elem->node_ptr(_i);
+  _node_to_weighted_tangential_velocity[node] += _test[_i][_qp] * _qp_tangential_velocity;
+  //  Moose::out << "Node locations: x: " << _lower_secondary_elem->node_ptr(_i)->get_info() <<
+  //  "\n"; Moose::out << "_node_to_weighted_tangential_velocity[node].size(): "
+  //             << _node_to_weighted_tangential_velocity.size() << "\n";
+  //  Moose::out << "qp_tangential_vel: " << MetaPhysicL::raw_value(_qp_tangential_velocity) <<
+  //  "\n"; Moose::out << "_secondary_x_dot[_qp]: " << MetaPhysicL::raw_value(_secondary_x_dot[_qp])
+  //  << "\n";
+}
+
+void
+ComputeFrictionalForceLMMechanicalContact::residualSetup()
+{
+  // Clear _node_to_weighted_gap map
+  ComputeWeightedGapLMMechanicalContact::residualSetup();
+
+  _node_to_weighted_tangential_velocity.clear();
```

Comment:
Ok good! This was going to be my first hypothesis for why you were getting large sparsity patterns ... if you weren't clearing your containers and your mesh was moving a lot theoretically you could end up with a derivatives container holding the whole interface mesh! (Although then your Newton solves probably wouldn't have converged because your SMP Jacobian would have been so bad ... or wait actually they would have been fine ... just your residuals/physics wouldn't have been right :smile: )